lib/evmone/analysis.cpp

```diff
@@ -177,31 +170,21 @@ code_analysis analyze(
         case OP_PC:
             instr.arg.p.number = static_cast<int>(code_pos - code - 1);
             break;
+        }
 
-        case OP_LOG0:
-        case OP_LOG1:
-        case OP_LOG2:
-        case OP_LOG3:
-        case OP_LOG4:
-            // TODO: This is not needed, but we keep it
-            //       otherwise compiler will not use the jumptable for switch implementation.
-            instr.arg.p.number = opcode - OP_LOG0;
-            break;
-
-        case OP_JUMP:
-        case OP_JUMPI:
-        case OP_STOP:
-        case OP_RETURN:
-        case OP_REVERT:
-        case OP_SELFDESTRUCT:
-            block = nullptr;
-            break;
+        if (create_new_block || (code_pos != code_end && *code_pos == OP_JUMPDEST))
```

Comment:
Maybe you could set `create_new_block` to `true` in `if (opcode == OP_JUMPDEST)` above, then this would be simpler

Reply:
That's right, but if `create_new_block` is already `true` we don't have to perform other checks.

Reply:
Renamed that to `is_terminator`. This is consistent with future changes in #158.